src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/AllocationService.kt

```diff
@@ -49,7 +49,7 @@ class AllocationService(
     val allocation = toOffenderProgramProfile(courseActivityId, request)
     val created = allocation.offenderProgramReferenceId == 0L
     return offenderProgramProfileRepository.save(allocation)
-      .let { UpsertAllocationResponse(it.offenderProgramReferenceId, created) }
+      .let { UpsertAllocationResponse(it.offenderProgramReferenceId, created, it.prison?.id ?: "") }
```

Comment:
Return the prisonId in the response so that nomis-update can include the prison on its telemetry. This is needed where DPS don't return the prison in their response which we would use if available.